web/app.js

```diff
@@ -670,18 +670,28 @@ const PDFViewerApplication = {
 
       // Enable dragging-and-dropping a new PDF file onto the viewerContainer.
       appConfig.mainContainer.addEventListener("dragover", function (evt) {
-        evt.preventDefault();
-
-        evt.dataTransfer.dropEffect =
-          evt.dataTransfer.effectAllowed === "copy" ? "copy" : "move";
+        const { items } = evt.dataTransfer;
+        for (const item of items) {
+          if (item.type === "application/pdf") {
+            evt.dataTransfer.dropEffect =
+              evt.dataTransfer.effectAllowed === "copy" ? "copy" : "move";
+            evt.preventDefault();
+            evt.stopPropagation();
+            return;
+          }
+        }
       });
       appConfig.mainContainer.addEventListener("drop", function (evt) {
-        evt.preventDefault();
-
         const { files } = evt.dataTransfer;
-        if (!files || files.length === 0) {
+        if (
+          !files ||
+          files.length === 0 ||
```

Comment:
Nit: Since you're touching this anyway, I'd just combine these checks.
```suggestion
          !files?.length ||
```

Reply:
You can combine even more, to `if (files?.[0].type !== "application/pdf")`